grpc/src/main/java/com/linecorp/armeria/server/grpc/ArmeriaServerCall.java

```diff
@@ -564,7 +564,7 @@ private void invokeHalfClose() {
             assert listener != null;
             listener.onHalfClose();
         } catch (Throwable t) {
-            close(t, new Metadata());
+            close(t, generateMetadataFromThrowable(t));
```

Comment:
How about introducing `close(Throwable)` method and replace all `close(t, new Metadata())` with the new one?